### PR TITLE
Add a new table `kernel keys` to the Linux platform

### LINKED ISSUE #7765: Table Request: `/proc/keys` on linux
# Feature request

I recently learned of linux's `/proc/keys`, and it seems like the sort of thing we should have a table for. 

[keyrings.7](https://man7.org/linux/man-pages/man7/keyrings.7.html) says:
> This file exposes a list of the keys for which the reading thread has view permission, providing various information about each key.  The thread need not possess the key for it to be visible in this file.

I am not sure if that matters here, or if there's an implied `uid` column.

Docs also mention `/proc/key-users` which might be interesting.

### PR DESCRIPTION
**Pull request by iko1**
This PR adds a new table called kernel_keys for Linux. This table exposes the content of the file /proc/keys,
this file exposes a list of the keys for which the reading thread has view permission, providing various information about each key. 

Fixes #7765

### CODE REVIEW

File: osquery/tables/system/linux/kernel_keys.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+static const std::string kProcKeysPath = "/proc/keys";
+
+QueryData genKernelKeys(QueryContext& context) {
+  if (!pathExists(kProcKeysPath).ok()) {
+    VLOG(1) << "Cannot find proc keys file: " << kProcKeysPath;
+    return {};
+  }
+
+  std::ifstream fd(kProcKeysPath, std::ios::in);
```
Comment (directionless) [inline osquery/tables/system/linux/kernel_keys.cpp:29]: This looks pretty clean, but please use the `read_file` helper.

**Review by directionless — CHANGES_REQUESTED** [review]

File: osquery/tables/system/linux/kernel_keys.cpp (hunk 1 above)
Comment (directionless) [reply osquery/tables/system/linux/kernel_keys.cpp:29]: https://github.com/osquery/osquery/blob/f15b05a48c43f2fa67609b5f47a7468a1bffdf1e/osquery/filesystem/filesystem.cpp#L226

Comment (iko1) [reply osquery/tables/system/linux/kernel_keys.cpp]: Fixed, Thanks for the comment

File: osquery/tables/system/linux/kernel_keys.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    VLOG(1) << "Error reading keyrings file: " << kProcKeysPath;
+    return {};
+  }
+
+  QueryData results;
+  for (const auto& key : osquery::split(content, "\n")) {
+    Row r = {};
+    auto details = osquery::split(key, " ");
+    if (details.size() != 10) {
+      // Interesting error case, the key line is not well formed.
```
Comment (directionless) [inline osquery/tables/system/linux/kernel_keys.cpp:41]: Does this merit a verbose warning?

File: specs/linux/kernel_keys.table [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+schema([
+    Column("serial_number", TEXT, "The serial key of the key."),
+    Column("flags", TEXT, "A set of flags describing the state of the key."),
+    Column("usage", BIGINT, "the number of threads and open file references that"
+                     "refer to this key."),
+    Column("timeout", TEXT, "The amount of time until the key will expire,"
+                     "expressed in human-readable form. The string perm here"
+                     "means that the key is permanent (no timeout).  The"
+                     "string expd means that the key has already expired."),
+    Column("permissions", TEXT, "The key permissions, expressed as four hexadecimal"
```
Comment (directionless) [inline specs/linux/kernel_keys.table:12]: It'll be interesting seeing if this field proves usable, of if we can improve it.

**Review by directionless — DISMISSED** [review]
I think this is mergable, so I'm approving. I think we could add a verbose message

File: osquery/tables/system/linux/kernel_keys.cpp (hunk 2 above)
Comment (iko1) [reply osquery/tables/system/linux/kernel_keys.cpp]: Indeed, added a verbose warning instead of this comment.

**Review by directionless — APPROVED** [review]

**Pull request merged**